doc/adr/0008-concept-for-handling-ldml.de-in-backend.md

```diff
@@ -0,0 +1,53 @@
+# 6. Concept for handling LDML.de in the backend
+
+Date: 2024-04-12
+
+## Status
+
+Proposed
+
+## Context
+
+We have identified two main issues with our current approach in the backend source code:
+- The current domain model is quite artificial, depicting just one specific use case (a.k.a amending laws).
+- The database model contains data that is duplicated, meaning present within the XML (xml columns) and in a separate column.
+- LDML.de is not the source of truth
+
+This ADR outlines a general concept for how to handle the XMLs in LDML.de to tackle the mentioned issues above.
+
+## Decision
+
+This concept consists of:
+1. The database model will be re-designed. We won't have anymore `amending_law`, `article` and `target_law` but rather one table for process data
+and another table for the work data. The latter will contain two column (`eli` and `xml`).
+2. The `xml` column of the work data table will be of type `TEXT`. We decided against using the xml type, at least for now. If we realize that we do
+a lot of filtering on the database level using specific nodes, we may migrate to xml type or also maybe use a materialized view.
+3. The mapper class mapping between the DTO and the domain model will be in charge of parsing the retrieved XML into a `w3c.dom.Document`.
+4. For the work data the domain model will consist of a wrapper class around the `w3c.dom.Document` class with getters for extracting using XPath
```

Comment:
Maybe have two sentences to make it less complex:
"For the work data the domain model will consist of a wrapper class around the `w3c.dom.Document` class with getters for specific data within the XML (e.g. the title, or the announcement date, etc.). We expect the getters to use XPath expressions on the XML."